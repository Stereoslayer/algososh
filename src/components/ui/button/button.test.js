import {Button} from './button';
import renderer from 'react-test-renderer';
import {render, screen, fireEvent} from '@testing-library/react';

describe('Button component test', () => {
    it('Button without text is rendered correctly', () => {
        const button = renderer.create(<Button/>).toJSON();
        expect(button).toMatchSnapshot();
    })
    it('Button with text is rendered correctly', () => {
        const props = {
            text: 'someText'
        };
        render(<Button {...props} />);
        const button = renderer.create(<Button {...props} />).toJSON();
        expect(screen.getByRole('button')).toHaveTextContent('someText');
        expect(button).toMatchSnapshot();
    })
    it('disabled Button is rendered correctly', () => {
        const props = {
            disabled: true
        };
        render(<Button {...props} />);
        const button = renderer.create(<Button {...props} />).toJSON();
        expect(screen.getByRole('button')).toBeDisabled();
        expect(button).toMatchSnapshot();
    })
    it('loading Button is rendered correctly', () => {
        const props = {
            isLoader: true
        };
        render(<Button {...props} />);
        const button = renderer.create(<Button {...props} />).toJSON();
        expect(screen.getByAltText('Загрузка.')).not.toEqual(null);
        expect(button).toMatchSnapshot();
    })
    it('callback is executed when the Button is pressed', () => {
        const callback = jest.fn();
        render(<Button onClick={callback}/>);
        const button = screen.getByRole('button');
        fireEvent.click(button);
        expect(callback).toHaveBeenCalled();
    })
})