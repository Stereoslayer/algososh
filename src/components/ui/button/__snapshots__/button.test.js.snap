// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component test Button with text is rendered correctly 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    someText
  </p>
</button>
`;

exports[`Button component test Button without text is rendered correctly 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button component test disabled Button is rendered correctly 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button component test loading Button is rendered correctly 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button test Button loading test 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button test Button render test 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button test Button with text test 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    someText
  </p>
</button>
`;

exports[`Button test disabled Button test 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
